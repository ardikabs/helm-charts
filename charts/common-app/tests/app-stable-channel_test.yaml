suite: app stable channel test
templates:
  - "deployment.yaml"
  - "service.yaml"
  - "ingress.yaml"
  - "hpa.yaml"
  - "pdb.yaml"
values:
  - "values/full_app-stable-channel.yaml"
release:
  name: podinfo-stable
  namespace: testing
tests:
  - it: should create Deployment
    template: deployment.yaml
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -stable$
      - equal:
          path: spec.template.spec.containers[?(@.name == "main")].image
          value: ghcr.io/ardikabs/etc/podinfo:6.3.3
  - it: should create 2 types of Service
    template: service.yaml
    asserts:
      - isKind:
          of: Service
      - matchRegex:
          path: metadata.name
          pattern: -stable$
        documentIndex: 0
      - equal:
          path: metadata.name
          value: podinfo
        documentIndex: 1
      - hasDocuments:
          count: 2
      - equal:
          path: spec.selector
          value:
            app.kubernetes.io/name: podinfo
            app.kubernetes.io/channel: stable
        documentIndex: 0
      - equal:
          path: spec.selector
          value:
            app.kubernetes.io/name: podinfo
        documentIndex: 1
      - matchSnapshot:
          path: spec
  - it: should create an Ingress
    template: ingress.yaml
    asserts:
      - isKind:
          of: Ingress
      - matchRegex:
          path: metadata.name
          pattern: -stable$
      - equal:
          path: spec.ingressClassName
          value: nginx
      - equal:
          path: spec.rules[*].host
          value: podinfo.ardikabs.com
  - it: should create a HPA
    template: hpa.yaml
    asserts:
      - isKind:
          of: HorizontalPodAutoscaler
      - isAPIVersion:
          of: autoscaling/v2
      - matchRegex:
          path: metadata.name
          pattern: -stable$
      - matchSnapshot:
          path: spec
  - it: should create a PDB
    template: pdb.yaml
    asserts:
      - isKind:
          of: PodDisruptionBudget
      - isAPIVersion:
          of: policy/v1
      - matchRegex:
          path: metadata.name
          pattern: -stable$
      - matchSnapshot:
          path: spec
