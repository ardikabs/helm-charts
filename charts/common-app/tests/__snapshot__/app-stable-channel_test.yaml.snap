should create 2 types of Service:
  1: |
    ports:
      - name: http
        port: 80
        protocol: TCP
        targetPort: http
    selector:
      app.kubernetes.io/channel: stable
      app.kubernetes.io/name: podinfo
    type: ClusterIP
  2: |
    ports:
      - name: http
        port: 80
        protocol: TCP
        targetPort: http
    selector:
      app.kubernetes.io/name: podinfo
    type: ClusterIP
should create a HPA:
  1: |
    maxReplicas: 3
    metrics:
      - resource:
          name: cpu
          target:
            averageUtilization: 70
            type: Utilization
        type: Resource
      - resource:
          name: memory
          target:
            averageUtilization: 70
            type: Utilization
        type: Resource
    minReplicas: 1
    scaleTargetRef:
      apiVersion: apps/v1
      kind: Deployment
      name: podinfo-stable
should create a PDB:
  1: |
    maxUnavailable: 25%
    selector:
      matchLabels:
        app.kubernetes.io/channel: stable
        app.kubernetes.io/name: podinfo
